examples/text-generation/run_generation.py

```diff
@@ -301,9 +301,9 @@ def setup_parser(parser):
     parser.add_argument(
         "--parallel_strategy",
         type=str,
-        choices=["tp", "none"],  # Add other strategies as needed
+        choices=["tp", "ep", "none"],  # Add other strategies as needed
         default="none",
-        help="Run multi card with the specified parallel strategy. Choices are 'tp' for Tensor Parallel Strategy or 'none'.",
+        help="Run multi card with the specified parallel strategy. Choices are 'tp' for Tensor Parallel Strategy or 'ep' for Expert Parallel Strategy or 'none'.",
```

Comment:
1) MOE models usually use EP since the main weight is usually in experts which are activated sparsely, TP doesn't handle well for these sparse activation cases. And since non-moe models don't have "experts", so yes, EP only for MOE models. 
2) When TP is set for this model, we will exit by showing below error message:
   "TypeError: DeepseekV2ForCausalLM.__init__() got an unexpected keyword argument 'parallel_strategy'"